java/src/Ice/src/main/java/com/zeroc/Ice/_ObjectPrxI.java

```diff
@@ -13,8 +13,7 @@
 
 /** Concrete proxy implementation. */
 public class _ObjectPrxI implements ObjectPrx, java.io.Serializable {
-  // TODO: Only public for 'IceInternal.ProxyFactory', should become protected after refactoring.
-  public _ObjectPrxI(Reference ref) {
+  protected _ObjectPrxI(Reference ref) {
```

Comment:
You should move the protected constructors after the public methods.

Reply:
Putting constructors before other methods is a _very_ strong Java convention.
Even if those constructors are less visible than the methods that follow.

ie:

1.) Public constructors
2.) Protected constructors
3.) Private constructors

4.) Public methods
5.) Protected methods
6.) Private methods

Reply:
... Well, some people will flip these and do methods first, then constructors.
(Although that's less common to see).

But the important part is that constructors should not be mixed in with methods, even if strictly following visibility ordering would say so.